### ISSUE TITLE
zero byte in middle of binary array

### ISSUE THREAD

**Issue by tskenb** [GeoffMontee/tds_fdw#38]
When there is a zero byte in the middle of a binary array postgresql interprets it as end-of-string in BuildTupleFromCStrings, so the rest of the array gets cut off.

This issue might be resolved as part of [issue 37](https://github.com/GeoffMontee/tds_fdw/issues/37), in which case I'll close this one.  But in case it isn't I'm opening this so it doesn't get forgotten.

**Comment by tskenb** [comment]
Interestingly, if my input value (in SQL Server) is a binary(n) or varbinary(n) then I get the error I described before, but if it is a varbinary(max) then the srctype comes in as SYBIMAGE, the bytes are converted to an actual string, which is then converted to a hex value for each ascii character.  So \xFA00BC comes back as \x666130306263, since \x66 is hex for 'F', etc.

I still expect this to be fixed as part of [issue 37](https://github.com/GeoffMontee/tds_fdw/issues/37), and I'll include SYBIMAGE in the case selection.

**Comment by GeoffMontee** [comment]
After pushing the commit for issue #37, this does appear to be fixed.

Created a table on MS SQL Server:

```
CREATE TABLE dbo.binary (
	id int primary key,
	value binary(4)
);

INSERT INTO dbo.binary (id, value) VALUES (1, 0x01020304);
```

Then in PostgreSQL:

```
CREATE FOREIGN TABLE binary4 (
	id int,
	value bytea
)
	SERVER mssql_svr
	OPTIONS (table 'dbo.binary', row_estimate_method 'showplan_all');	
```

The results look good:

```
postgres=# SELECT * FROM binary4;
 id |   value
----+------------
  1 | \x01020304
(1 row)
```

**Comment by tskenb** [comment]
Actually, the constant needs to have two zeros in a row, since each pair is a byte - e.g., \x01020004.  I did a similar test and found that it was indeed fixed.

**Comment by GeoffMontee** [comment]
Ah, of course! My mistake.

I'm glad it's fixed.

**Comment by GeoffMontee** [comment]
Fixed in 1.0.6.
